### ISSUE TITLE
Feature Request: Search Building as a waypoint

### ISSUE THREAD

**Issue by freghar** [zen-mod/ZEN#502]
**Is your feature request related to a problem?**
The current implementation works well enough, but a new waypoint type would allow better chaining to a more complex patrol path (ie. searching several buildings in a row and moving around the area, before CYCLE and doing it all again).

**Solution you'd like:**
SEARCH BUILDING waypoint type that executes `fnc_searchBuilding.sqf` on a building nearest to `waypointPosition` of the WP and either completes once the search is done or otherwise ensures further waypoints wait until the search is complete (ie. by inserting a dummy WP, `addWaypoint` with `index`).

**Alternatives you've considered:**
Achilles has such functionality, though the chaining was always flaky. The obvious alternative is wasting the GM's time by waiting around until the `AWARE` group moves close enough, because the current `Search Building` switches a group to `COMBAT`, making long-distance approaches infeasible. 😄

**Comment by freghar** [comment]
> further waypoints wait until the search is complete

This seems to be already covered by `fnc_searchBuilding.sqf`:
```sqf
// Prevent the group from completing waypoints while the building is being searched
[QEGVAR(common,lockWP), [_group, true], _group] call CBA_fnc_targetEvent;
```
